### PR TITLE
STORM-3534 Add generic resources to UI

### PR DESCRIPTION
**Pull request by dandsager1**

### CODE REVIEW

**Comment by dandsager1** [comment]
Added Screen Shots:

#Main Page#
startPage

#Topology Page#
topologyPage

#Component Page#
componentPage

File: storm-server/src/main/java/org/apache/storm/scheduler/resource/normalization/NormalizedResourceRequest.java [hunk 1]
```diff
@@ -178,6 +178,43 @@ private static void putIfMissing(Map<String, Double> dest, String destKey, Map<S
         return ret;
     }
 
+    /*
+     * return map with non generic resources removded
```
Comment (agresch) [inline storm-server/src/main/java/org/apache/storm/scheduler/resource/normalization/NormalizedResourceRequest.java]: spelling of removded

File: storm-server/src/main/java/org/apache/storm/scheduler/resource/normalization/NormalizedResourceRequest.java [hunk 2]
```diff
@@ -178,6 +178,43 @@ private static void putIfMissing(Map<String, Double> dest, String destKey, Map<S
...
+     */
+    public static void filterGenericResources(Map<String, Double> map) {
```
Comment (agresch) [inline storm-server/src/main/java/org/apache/storm/scheduler/resource/normalization/NormalizedResourceRequest.java]: can we rename this removeNonGenericResources() or filterOutNonGenericResources() or something?  I think that would be clearer.

File: storm-server/src/main/java/org/apache/storm/scheduler/resource/normalization/NormalizedResourceRequest.java (hunk 1 above)
Comment (dandsager1) [reply storm-server/src/main/java/org/apache/storm/scheduler/resource/normalization/NormalizedResourceRequest.java]: fixed

File: storm-server/src/main/java/org/apache/storm/scheduler/resource/normalization/NormalizedResourceRequest.java (hunk 2 above)
Comment (dandsager1) [reply storm-server/src/main/java/org/apache/storm/scheduler/resource/normalization/NormalizedResourceRequest.java]: renamed

**Pull request merged**